File: app/src/main/java/com/davismiyashiro/expenses/model/localrepo/TabsCursorWrapper.java
package com.davismiyashiro.expenses.model.localrepo;

import android.database.Cursor;
import android.database.CursorWrapper;

import com.davismiyashiro.expenses.datatypes.Expense;
import com.davismiyashiro.expenses.datatypes.Participant;
import com.davismiyashiro.expenses.datatypes.ReceiptItem;
import com.davismiyashiro.expenses.datatypes.Split;
import com.davismiyashiro.expenses.datatypes.Tab;

import static com.davismiyashiro.expenses.model.localrepo.TabDbSchema.*;

/**
 * Adds wrapper methods to make it easier to retrieve data from the cursor
 */
public class TabsCursorWrapper extends CursorWrapper{

    /**
     * Creates a cursor wrapper.
     *
     * @param cursor The underlying cursor to wrap.
     */
    public TabsCursorWrapper(Cursor cursor) {
        super(cursor);
    }

    /**
     * Returns object from DB column
     *
     * @return
     */
    public Tab getTab () {
        String  uuid = getString(getColumnIndex(TabTable.UUID));
        String tabName = getString(getColumnIndex(TabTable.GROUPNAME));

        return new Tab (uuid, tabName);
    }

    public Participant getParticipant() {
        String uuid = getString(getColumnIndex(ParticipantTable.UUID));
        String partName = getString(getColumnIndex(ParticipantTable.NAME));
        String partEmail = getString(getColumnIndex(ParticipantTable.EMAIL));
        String partNumber = getString(getColumnIndex(ParticipantTable.NUMBER));
        String tabId = getString(getColumnIndex(ParticipantTable.TAB_ID));

        return Participant.Companion.retrieveParticipant(uuid, partName, partEmail, partNumber, tabId);
    }

    public Expense getExpense() {
        String expId = getString(getColumnIndex(ExpenseTable.UUID));
        String expDesc = getString(getColumnIndex(ExpenseTable.DESCRIPTION));
        double expValue = getDouble(getColumnIndex(ExpenseTable.VALUE));
        String expTabId = getString(getColumnIndex(ExpenseTable.TAB_ID));

        return Expense.Companion.retrieveExpense(expId, expDesc, expValue, expTabId);
    }

    public Split getSplit() {
        String splitId = getString(getColumnIndex(SplitTable.UUID));
        String partId = getString(getColumnIndex(SplitTable.PARTICIPANT_ID));
        String expId = getString(getColumnIndex(SplitTable.EXPENSE_ID));
        double splitVal = getDouble(getColumnIndex(SplitTable.SPLIT_VAL));
        String splitTabId = getString(getColumnIndex(SplitTable.TAB_ID));

        return Split.Companion.retrieveSplit(splitId, partId, expId, 0, splitVal, splitTabId);
    }

    public ReceiptItem getReceipItem() {
        String partId = getString(getColumnIndex(SplitTable.PARTICIPANT_ID));
        String partName = getString(getColumnIndex(ParticipantTable.NAME));
        String expDesc = getString(getColumnIndex(ExpenseTable.DESCRIPTION));
        double splitVal = getDouble(getColumnIndex(SplitTable.SPLIT_VAL));

        return new ReceiptItem(partId, partName, expDesc, splitVal);
    }
}

File: app/src/main/java/com/davismiyashiro/expenses/model/localrepo/TabsCursorWrapper.kt
package com.davismiyashiro.expenses.model.localrepo

import android.database.Cursor
import android.database.CursorWrapper

import com.davismiyashiro.expenses.datatypes.Expense
import com.davismiyashiro.expenses.datatypes.Participant
import com.davismiyashiro.expenses.datatypes.ReceiptItem
import com.davismiyashiro.expenses.datatypes.Split
import com.davismiyashiro.expenses.datatypes.Tab

import com.davismiyashiro.expenses.model.localrepo.TabDbSchema.*

/**
 * Adds wrapper methods to make it easier to retrieve data from the cursor
 */
class TabsCursorWrapper(cursor: Cursor) : CursorWrapper(cursor) {

    /**
     * Returns object from DB column
     *
     * @return
     */
    val tab: Tab
        get() {
            val uuid = getString(getColumnIndex(TabTable.UUID))
            val tabName = getString(getColumnIndex(TabTable.GROUPNAME))

            return Tab(uuid, tabName)
        }

    val participant: Participant
        get() {
            val uuid = getString(getColumnIndex(ParticipantTable.UUID))
            val partName = getString(getColumnIndex(ParticipantTable.NAME))
            val partEmail = getString(getColumnIndex(ParticipantTable.EMAIL))
            val partNumber = getString(getColumnIndex(ParticipantTable.NUMBER))
            val tabId = getString(getColumnIndex(ParticipantTable.TAB_ID))

            return Participant.retrieveParticipant(uuid, partName, partEmail, partNumber, tabId)
        }

    val expense: Expense
        get() {
            val expId = getString(getColumnIndex(ExpenseTable.UUID))
            val expDesc = getString(getColumnIndex(ExpenseTable.DESCRIPTION))
            val expValue = getDouble(getColumnIndex(ExpenseTable.VALUE))
            val expTabId = getString(getColumnIndex(ExpenseTable.TAB_ID))

            return Expense.retrieveExpense(expId, expDesc, expValue, expTabId)
        }

    val split: Split
        get() {
            val splitId = getString(getColumnIndex(SplitTable.UUID))
            val partId = getString(getColumnIndex(SplitTable.PARTICIPANT_ID))
            val expId = getString(getColumnIndex(SplitTable.EXPENSE_ID))
            val splitVal = getDouble(getColumnIndex(SplitTable.SPLIT_VAL))
            val splitTabId = getString(getColumnIndex(SplitTable.TAB_ID))

            return Split.retrieveSplit(splitId, partId, expId, 0, splitVal, splitTabId)
        }

    val receiptItem: ReceiptItem
        get() {
            val partId = getString(getColumnIndex(SplitTable.PARTICIPANT_ID))
            val partName = getString(getColumnIndex(ParticipantTable.NAME))
            val expDesc = getString(getColumnIndex(ExpenseTable.DESCRIPTION))
            val splitVal = getDouble(getColumnIndex(SplitTable.SPLIT_VAL))

            return ReceiptItem(partId, partName, expDesc, splitVal)
        }
}
